provider/source/src/units/data.rs

```diff
@@ -126,17 +104,25 @@ impl crate::IterableDataProviderCached<UnitsDisplayNameV1Marker> for SourceDataP
         ) -> Result<(), DataError> {
             let quantities = length_patterns
                 .keys()
-                // TODO: remove this filter once we are supporting all the units categories.
-                // NOTE:
-                //  if this filter is removed, we have to add a filter to remove all the prefixes.
+                .filter(|&key| {
+                    !key.starts_with(|c: char| c.is_ascii_digit())
+                        && !["per", "times", "power"]
```

Comment:
nit: add comment explaining why these prefixes are excluded

Reply:
I added in the other filter